src/main/java/org/broadinstitute/hellbender/tools/sv/cluster/CanonicalSVLinkage.java

```diff
@@ -222,12 +274,31 @@ private static int getLength(final SVCallRecord record, final int missingInserti
             return record.getLength() == null ? missingInsertionLength : Math.max(record.getLength(), 1);
         } else if (record.getType() == GATKSVVCFConstants.StructuralVariantAnnotationType.BND) {
             return record.getPositionB() - record.getPositionA() + 1;
+        } else if (record.getType() == GATKSVVCFConstants.StructuralVariantAnnotationType.CPX) {
+            return getLengthComplex(record, missingInsertionLength);
         } else {
             // TODO lengths less than 1 shouldn't be valid
             return Math.max(record.getLength() == null ? 1 : record.getLength(), 1);
         }
     }
 
+    /**
+     * Returns length to be used for complex variant matching
+     */
+    private static int getLengthComplex(final SVCallRecord record, final int missingInsertionLength) {
```

Comment:
It would be good to explicitly unit-test this instead of just testing the methods that call it

Reply:
Agreed but it's deleted now